src/main/util/WeightedGraph.java

```diff
@@ -0,0 +1,91 @@
+package util;
+
+import java.util.HashSet;
+import java.util.function.BiFunction;
+
+public class WeightedGraph<V, W extends Comparable<W>> {
```

Comment:
```suggestion
public class CompleteWeightedGraph<V, W extends Comparable<W>> {
```
The fact that it's complete is quite important to the way the interface is structured (eg the fact that you can't add edges)